controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentMetricsMaintainer.java

```diff
@@ -36,22 +37,24 @@ protected void maintain() {
                     DeploymentMetrics appMetrics = new DeploymentMetrics(metrics.queriesPerSecond(), metrics.writesPerSecond(),
                                                                          metrics.documentCount(), metrics.queryLatencyMillis(), metrics.writeLatencyMillis());
 
-                    // Avoid locking for a long time, due to slow YAMAS.
                     try (Lock lock = controller().applications().lock(application.id())) {
-                        // Deployment (or application) may have changed (or be gone) now:
-                        controller().applications().get(application.id()).ifPresent(freshApplication -> {
-                            Deployment freshDeployment = freshApplication.deployments().get(deployment.zone());
-                            if (freshDeployment != null)
-                                controller().applications().store(freshApplication.with(freshDeployment.withMetrics(appMetrics)), lock);
-                        });
+
+                        // Deployment or application may have changed (or be gone) now:
+                        application = controller().applications().require(application.id());
+                        if (application == null)
+                            break;
+
+                        deployment = application.deployments().get(deployment.zone());
+                        if (deployment == null)
+                            continue;
+
+                        controller().applications().store(application.with(deployment.withMetrics(appMetrics)), lock);
                     }
                 }
                 catch (UncheckedIOException e) {
-                    log.warning("Timed out talking to YAMAS; retrying in " + maintenanceInterval() + ":\n" + e);
+                    log.log(Level.WARNING, "Timed out talking to YAMAS; retrying in " + maintenanceInterval() + ":\n", e);
```

Comment:
Also lose the ":\n". Those are not for you to add  :-)

Reply:
Right, I guess the logger does that with the .log(...), then.